### PR TITLE
Handle font licenses when editing theme metadata

### LINKED ISSUE #525: Font licenses are no longer automated with new Core Font Library functionality
Part of replacing the integrated font management with the new Font LIbrary in Core means that we're losing the automated font license functionality provided by this plugin.

This is currently handled with the `manage_font_license()` function, included [here](https://github.com/WordPress/create-block-theme/blob/trunk/admin/class-manage-fonts.php#L335).

Perhaps this could be added as a follow-up to https://github.com/WordPress/create-block-theme/pull/518 and https://github.com/WordPress/create-block-theme/pull/519.

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What
- Handle font licenses in the frontend 'edit theme metadata'.
- Adds a button to get theme fonts license text info updated.

## Why?
To avoid the need of collecting the font licesing information manually.
- This is part of the effort to re-introduce the font license theme automation (
Font licenses are no longer automated with new Core Font Library functionality #525).

## How
Calls the /font-families/ endpoint and process the data to create the fille the font credits text area.
Using a button to get the latest licenses has 2 benefits:
- Font assets metadata are loaded only when it is necessary.
- The theme author can add extra text to the one auto generated.

:warning: This PR depends on https://github.com/WordPress/create-block-theme/pull/648 being merged

## Screencast
This video features how the font licences can be updated for a theme.

[Screencast from 30-05-24 12:00:58.webm](https://github.com/WordPress/create-block-theme/assets/1310626/cffa1751-3491-48be-981b-a57955aa0160)

### CODE REVIEW

File: src/resolvers.js [hunk 1]
```diff
@@ -85,3 +85,20 @@ export async function downloadExportedTheme() {
...
+			path: '/create-block-theme/v1/font-families',
+			method: 'GET',
+			headers: {
+				'Content-Type': 'application/json',
+			},
+		} );
+		return response.data;
+	} catch ( e ) {
+		// eslint-disable-next-line no-console
+		console.error( e );
```
Comment (vcanales) [inline src/resolvers.js]: Remove console statement

```suggestion
		
```

File: src/utils/fonts.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+		return {
+			fontName: nameTable.get( 16 ) || nameTable.get( 1 ),
+			copyright: nameTable.get( 0 ),
+			source: nameTable.get( 11 ),
+			license: nameTable.get( 13 ),
+			licenseURL: nameTable.get( 14 ),
+		};
+	} catch ( error ) {
+		// eslint-disable-next-line no-console
+		console.error( 'Failed to get font license for font file.', error );
```
Comment (vcanales) [inline src/utils/fonts.js]: Remove console statement

```suggestion
		
```

File: src/utils/fonts.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+	try {
+		// Get the license from the font face url.
+		return await getFontFileLicenseFromUrl( faceUrl );
+	} catch ( error ) {
+		// eslint-disable-next-line no-console
+		console.error(
+			'Failed to get font license for font family: ',
+			fontFamily,
+			error
+		);
```
Comment (vcanales) [inline src/utils/fonts.js]: Remove console statement

```suggestion
		
```

**Review by vcanales — COMMENTED** [review]
I see there are a few `console.error` calls; I think we should implement proper error management instead.

File: src/editor-sidebar/metadata-editor-modal.js [hunk 4]
```diff
@@ -230,6 +239,45 @@ Plugin Description`,
...
+						</>
+					}
+					// eslint-disable-next-line @wordpress/i18n-no-collapsible-whitespace
+					placeholder={ __(
+						`Font Name
+Copyright
+License
+Source`,
+						'create-block-theme'
+					) }
```
Comment (vcanales) [inline src/editor-sidebar/metadata-editor-modal.js]: Translate each part individually instead.

```suggestion
					placeholder={ `${ __( 'Font Name', 'create-block-theme' ) }
${ __( 'Copyright', 'create-block-theme' ) }
${ __( 'License', 'create-block-theme' ) }
${ __( 'Source', 'create-block-theme' ) }` }
```

Also, perhaps modify to clarify how each part of the license should look like.

**Comment by matiasbenedetto** [comment]
I removed the error handling where it was not needed and added it in the function where the license text is requested. If getting the license fails, the UI will throw an alert. 


Until merging https://github.com/WordPress/create-block-theme/pull/648 this PR will display this error:

[Screencast from 23-05-24 12:20:13.webm](https://github.com/WordPress/create-block-theme/assets/1310626/62c8b45b-ca19-42ca-b534-38262d59026c)

Why an alert? Because the snackbar is behind the modal component and is not visible. As that problem is not easy to fix (reference: https://github.com/WordPress/gutenberg/issues/52609) I think for now the alert is good enough. 

[Screencast from 23-05-24 12:02:53.webm](https://github.com/WordPress/create-block-theme/assets/1310626/d84c0247-6b84-4600-9676-7f81f70a5194)

File: src/utils/fonts.js (hunk 2 above)
Comment (matiasbenedetto) [reply src/utils/fonts.js]: handled it here https://github.com/WordPress/create-block-theme/pull/649/commits/18839a3c37b6bff4cae725f81bd3b137c463c882

File: src/resolvers.js (hunk 1 above)
Comment (matiasbenedetto) [reply src/resolvers.js]: handled it here https://github.com/WordPress/create-block-theme/pull/649/commits/18839a3c37b6bff4cae725f81bd3b137c463c882

**Review by vcanales — APPROVED** [review]
Looking good, although you'll probably have to update the branch with `trunk`, since there's a new eslint rule enforcing the specification of which rule is being disabled when eslint-disable comments are used.

**Pull request merged**